model/project_parser.go

```diff
@@ -797,9 +800,21 @@ func GetProjectFromFile(ctx context.Context, opts GetProjectOpts) (ProjectInfo,
 // intermediate project representation (i.e. before selectors or
 // matrix logic has been evaluated).
 // If unmarshalStrict is true, use the strict version of unmarshalling.
-func createIntermediateProject(yml []byte, unmarshalStrict bool) (*ParserProject, error) {
+func createIntermediateProject(yml []byte, unmarshalStrict, useUpgradedYAML bool) (*ParserProject, error) {
 	p := ParserProject{}
-	if unmarshalStrict {
+	if useUpgradedYAML && unmarshalStrict {
```

Comment:
Should we add a TODO here + JIRA ticket for after the grace period?

Reply:
Forgot to respond to this earlier, but I'm gonna keep this ticket open until this temporary logic is removed, so I don't think we need another follow-up ticket.